src/components/CallView/Video.vue

```diff
@@ -159,16 +159,49 @@ export default {
 			return participantName
 		},
 
-		currentParticipant() {
+		participant() {
+			if (typeof this.token === 'undefined') {
+				return {
+					participantType: undefined,
+				}
+			}
+
+			if (!this.model.attributes.userId) {
+				return {
+					// The participant might be a GUEST_MODERATOR, but
+					// unfortunately it is not possible to know it.
+					participantType: PARTICIPANT.TYPE.GUEST,
+				}
+			}
+
+			const participantIndex = this.$store.getters.getParticipantIndex(this.token, { participant: this.model.attributes.userId })
+			if (participantIndex !== -1) {
+				return this.$store.getters.getParticipant(this.token, participantIndex)
+			}
+
+			return {
+				participantType: undefined,
+			}
+		},
+
+		isSendingMediaOrScreensharingAllowed() {
+			return this.participant.participantType === PARTICIPANT.TYPE.OWNER
```

Comment:
deja vu

I wonder if we should move this to a property and let the callview do this check and tell all sub-components what's allowed or not ?